lib/Frontend/CompilerInvocation.cpp

```diff
@@ -111,6 +111,49 @@ static void updateRuntimeLibraryPaths(SearchPathOptions &SearchPathOpts,
   }
 }
 
+static void setIRGenOutputOptsFromFrontendOptions(IRGenOptions &IRGenOpts,
```

Comment:
Using the pass-reference-in formulation allows information in the passed-in structure to propagate to the end-product. Is this causality flow required? If not, I suggest you just return a new` IRGenOptions` here.
If so, it would be great to explain in a comment when and what information needs to flow through.

Reply:
I'll document this.  In general, these kinds of things are layering violations - clearly IRGenOpts.OutputKind is a function of the FrontendOptions, but IRGen doesn't have access to that information.  I just want them to be formal layering violations if they must be here for now.

Reply:
In answer to your first question: It is required.  Think of this as a form of options parsing parsing!